lib/jobs/run-work-items.js

```diff
@@ -207,6 +208,37 @@ function runWorkItemsJobFactory(
             );
         });
     };
+    
+    /**
+     * @function _executeRedfishPoller
+     * @description execute the redfish worker item 
+     */
+    RunWorkItemsJob.prototype._executeRedfishPoller = function (workItem) {
```

Comment:
After examine this function,  I see it only does synchronous tasks, so for me, this async design is a little complicated or redundant.
Though `Promise` can be used to split the synchronous task to sub-tasks so it can be done in several tick-clock not once. But I still don't find there is a need to do this splitting as well.